src/components/uv-scroll.js

```diff
@@ -1,20 +1,79 @@
+const textureToData = new Map();
+const registeredTextures = [];
+
+export class UVScrollSystem {
```

Comment:
Not important to address here (and I know we have the same thing written elsewhere because I've been doing this too) but I think we should consider ending our practice of wrapping `tick` functions in a `class`es when there is no `constructor`, local data, or other `class`-construct helping us in some way. Our preference is something like, "Perform all the transformations related to a particular subsystem at a known point during the frame, and batch like-transformations instead of doing it at a per-entity level." We can stick to this preference without the "`class` with `tick` function" pattern. 

One thing we are missing is a uniform way of storing/accessing entity data, though this doesn't seem to cause us trouble very often in practice. Here (and elsewhere) we work around the problem by initializing/registering some entity data in an aframe component lifecycle method while making sure the the component-data stores `textureToData` and `registeredTextures` are available to the (singular) "system" `tick`.

Reply:
Yep I agree, there is not much practical benefit to this except that when you look in HubsSystems you dont have to wonder "Hey why is only the uv scroll system done like that?". I would rather just convert over the ones that make sense all at once. But then also that feels like a waste of time... So we continue doing the silly pattern.